.github/workflows/push_pr.yml

```diff
@@ -41,13 +47,52 @@ jobs:
           only-new-issues: true
 
   test:
-    name: test
+    name: Run unit tests
     runs-on: ubuntu-latest
-    needs: [build]
+    needs: [ build ]
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-go@v2
         with:
           go-version: '1.16'
+      - uses: actions/cache@v2
+        with:
+          path: ~/go/pkg/mod
+          key: gomod-${{ hashFiles('**/go.sum') }}
+          restore-keys: |
+            gomod-
       - name: Test
-        run: make test
+        run: make test-unit
+
+  test-e2e:
+    name: Run e2e/integration tests
+    runs-on: ubuntu-latest
+    needs: [ build ]
+    timeout-minutes: 10
+    strategy:
+      max-parallel: 4
+      fail-fast: false
+      matrix:
+        kubernetes-version: [ "v1.15.12", "v1.16.15", "v1.17.17", "v1.18.18", "v1.19.10", "v1.20.6", "v1.21.0" ]
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.16'
+      - name: Setup Minikube
+        uses: manusa/actions-setup-minikube@v2.4.0
+        with:
+          minikube version: v1.20.0
```

Comment:
keys with spaces, nice :see_no_evil:

Reply:
Yeah, it's pretty annoying to me as well :(